### PR TITLE
[docs] Fix vale warning about Latin abbrv., inconsistent case in headings and update verbiage

### PR DESCRIPTION
**Pull request by amandeepmittal**
# Why

In some of our docs, Vale complains about inconsistent use/not using sentence cases for section headings, wordiness, and Latin abbreviations.

# How

This PR fixes most of the inconsistent usage in section headings, wordiness, and Latin abbreviations.
- Also update the exception list in `HeadingCase.yml` to include keywords that I learned about after going through these docs manually. The guideline doesn't apply to them and that's why they are added to the exception list.
- _Note:_ There is also a fair amount of warnings that Vale throws. Some are related to using numbered lists in heading sections (will tackle this in another PR since it might require us to convert those sections into steps using `<Step>`.
  - Also, Vale throws warnings for content inside the hidden files such as **pages/versions/latest/sdk/..**`, which we can ignore at this time since as it doesn't seem to have a straightway mechanism for ignore specific sub directories or files (at least I could not find at the moment but will look into its config option).
- In some places, minor grammatical errors are also fixed.

# Test Plan

N/A

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: docs/pages/eas-update/introduction.mdx [hunk 1]
```diff
@@ -17,19 +17,19 @@ Still using Classic Updates? We moved those docs to [the archive](/archive/class
...
+### Broad compatibility
 
 EAS Update works with vanilla React Native apps. All you need to do is install the lightweight `expo` npm package as a prerequisite, making it a compelling alternative to solutions such as [Codepush](/eas-update/codepush).
 
 ### Tailored for Expo
 
 EAS Update is designed for Expo apps, making integration easy and workflows efficient. Use the website UIs to check app status and debug, especially for developers using EAS Build.
 
-### Customizable Update Strategies
+### Customizable Update strategies
```
Comment (Simek) [inline docs/pages/eas-update/introduction.mdx:32]:
```suggestion
### Customizable update strategies
```

Shouldn't "Update" also be lowercased in this sentence?

File: docs/pages/router/reference/search-parameters.mdx [hunk 2]
```diff
@@ -1,7 +1,7 @@
 ---
-title: Use Search Parameters
+title: Use Search parameters
```
Comment (Simek) [inline docs/pages/router/reference/search-parameters.mdx:2]:
```suggestion
title: Use search parameters
```

File: docs/pages/router/reference/search-parameters.mdx [hunk 3]
```diff
@@ -120,7 +120,7 @@ Pressing "Visit james" has a similar effect:
 
 </Step>
 
-## Statically-typed Search Parameters
+## Statically-typed Search parameters
```
Comment (Simek) [inline docs/pages/router/reference/search-parameters.mdx:123]:
```suggestion
## Statically-typed search parameters
```

**Review by Simek — APPROVED** [review]
Tested locally and verified diffs, left few minor casing nit, but beside those changeset LGTM! 👍

File: docs/pages/eas-update/introduction.mdx (hunk 1 above)
Comment (amandeepmittal) [reply docs/pages/eas-update/introduction.mdx:32]: Yes good catch!

**Pull request merged**